### PR TITLE
Ncnewst6

### PR DESCRIPTION
**Pull request by Joelymodevs**

### CODE REVIEW

File: src/Articles.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      })
+      .catch((err) => {
+        console.log(err);
+        setIsLoading(false);
+      });
+  }, []);
+
+  if (isLoading) {
+    return (
+      <div>
```
Comment (Augs0) [inline src/Articles.jsx]: This is an example of those unnecessary divs I mentioned :D

File: src/Articles.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        <p>Loading...</p>
+      </div>
+    );
+  }
+  return (
+    <div className="articles">
+      {articles.map((article) => (
+        <Link to={`/articles/${article.article_id}`} key={article.title}>
+        <div key={article.article_id} className="article-card">
```
Comment (Augs0) [inline src/Articles.jsx:31]: What might be better than a div?

File: src/FullArticle.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React, { useEffect, useState } from "react";
+import { getArticles } from "./utils/getAllArticles";
+import { getArticleById } from "./utils/getArticleById";
+import { useParams } from "react-router-dom";
+import './ArticlePage.css'
+import Comments from './comments'
+
+const FullArticle = () => {
+    const [article, setArticle] = useState(null)
```
Comment (Augs0) [inline src/FullArticle.jsx]: This should probably be an empty object to match the data we expect

File: src/comments.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        getCommentsById(id).then((data) => {
+            setComments(data)
+            setIsLoading(false);
+        }).catch((err) => {
+            console.log(err)
+            setIsLoading(false)
+        })
+    }, [id])
+    if (isLoading) {
+        return (
```
Comment (Augs0) [inline src/comments.jsx:18]: We handle loading but we don't handle 0 comments on an article. How could we handle this case as well?

File: src/utils/getAllArticles.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import axios from "axios";
```
Comment (Augs0) [inline src/utils/getAllArticles.js:1]: Your utils don't need to be in their own files. What if we have 100+ api calls? We can have a utils file where they all live :)

**Review by Augs0 — CHANGES_REQUESTED** [review]
Overall, a great start Joel. Please add some logic to handle an article with 0 comments as well as thinking about your elements in the HTML as you go forward.

File: src/Articles.jsx (hunk 2 above)
Comment (Joelymodevs) [reply src/Articles.jsx:31]: article? :)

Comment (Joelymodevs) [reply src/Articles.jsx:31]: article? :)

File: src/utils/getAllArticles.js (hunk 5 above)
Comment (Joelymodevs) [reply src/utils/getAllArticles.js:1]: noted

File: src/comments.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                <p>Comments Loading...</p>
+        )
+    } else if (!comments) {
+        return (
+            <p>No Comments To Be Found Here Amigo!</p>
```
Comment (Augs0) [inline src/comments.jsx:23]: Olé!

**Review by Augs0 — APPROVED** [review]
Well done, approved as changes all made :D

**Pull request closed without merging**